core/src/main/scala-3/org/wartremover/warts/PlatformDefault.scala

```diff
@@ -25,7 +25,7 @@ object PlatformDefault extends WartTraverser {
             t.asExpr match {
               case '{ ($x: String).getBytes } =>
                 error(tree.pos, getByte)
-              case '{ ($x1: String).toLowerCase } | '{ ($x2: String).toUpperCase } =>
+              case '{ ($_ : String).toLowerCase } | '{ ($_ : String).toUpperCase } =>
```

Comment:
Looks like the `${_}` syntax is incorrect in Scala 3.1.1, but allowed since 3.1.2. Aslo the `${_}` does violate the scalafmt rules which require it be defined as `${ _ }` (with whitespaces around _). 
As the compiler suggested I've replaced it with `$_` which does require additional whitespace before type descriptor. 
When testing locally it has compiled for all 3.1.x versions and 3.2.x nighlty. 
I've not replaced other definitions of `$x` with wildcards since even though variables are unused the code is better formated at the current form - we would need to replace `($x: String)` with `($_ : String)` and I don't really like this extra whitespace before the colon :)